controllers/networkfence_controller.go

```diff
@@ -113,37 +118,66 @@ func (r *NetworkFenceReconciler) Reconcile(ctx context.Context, req ctrl.Request
 		return ctrl.Result{}, err
 	}
 
+	nf := NetworkFenceInstance{
+		ctx:              ctx,
+		reconciler:       r,
+		logger:           logger,
+		instance:         nwFence,
+		controllerClient: client,
+	}
+
 	// check if the networkfence object is getting deleted, if yes, then send
 	// UnfenceCluster request to the driver.
 	// After the UnfenceCluster request is processed, the finalizer will be removed
 	// so that the networkfence object can be deleted gracefully.
-	if !nwFence.GetDeletionTimestamp().IsZero() {
+	if !nf.instance.GetDeletionTimestamp().IsZero() {
 		if util.ContainsInSlice(nwFence.GetFinalizers(), networkFenceFinalizer) {
 
-			err := r.unfenceClusterNetwork(ctx, nwFence, client, logger)
+			err := nf.deleteNetworkFenceResource()
```

Comment:
Call removefinalizer directly ? 

No need to Unfence here?

Reply:
Yes, no need to unfence here. Only reason to do it in a separate function to ensure that everything related to deletion of the resource is contained in a function. As of today with this PR, we are only removing the finalizer in the deletion handling. In future if something more needs to be done (cant think of any as of now though), then the function deleteNetworkFenceResource would be the one doing the additional things.

Let me know your thoughts.